SmartLogStatisticsTests/IntegrationTest/StatisticsTest.cs

```diff
@@ -0,0 +1,282 @@
+            _context.Log.AddRange(logLines);
+            _context.Firmware.AddRange(firmwares);
+            _context.SaveChanges();
+        }
+
+        /**
+         * TIS-27
+         * Verificare che l'ottenimento delle statistiche avvenga correttamente
+         */
+        [TestMethod()]
+        public void GoodStatisticsTest() {
+
+            PopulateContext();
+
+            OkObjectResult result = (OkObjectResult)_controller.Statistics(new DateTime(2022, 03, 04), new DateTime(2022, 07, 06));
+
+            Assert.IsNotNull(result.Value);
+
+            StatisticsDto val = (StatisticsDto)result.Value;
+
+            Assert.AreEqual(new DateTime(2022, 03, 04), val.StartDate);
+            Assert.AreEqual(new DateTime(2022, 07, 06), val.EndDate);
+
+            var stats = val.Statistics;
+            var fileNumb = stats.Find(stat => stat.Name == "Numero di file");
+            var maxEventNumb = stats.Find(stat => stat.Name == "Massimo numero di eventi");
+            var eventMean = stats.Find(stat => stat.Name == "Media di eventi");
+            var stdDeviation = stats.Find(stat => stat.Name == "Deviazione standard");
+
+            Assert.IsNotNull(fileNumb);
+            Assert.IsNotNull(maxEventNumb);
+            Assert.IsNotNull(eventMean);
+            Assert.IsNotNull(stdDeviation);
+
+            Assert.AreEqual(2, fileNumb.Value);
+            Assert.AreEqual(5, maxEventNumb.Value);
+            Assert.AreEqual(4, eventMean.Value);
+            Assert.AreEqual(1, stdDeviation.Value);
+
+        }
+
+        /**
+         * TIS-28
+         * Verificare che viene visualizzato un messagio di query vuota nel caso in cui l'ottenimento delle statistiche ritorni risultato vuoto
+         */
+        [TestMethod()]
+        public void NoStatisticsTest() {
+
+            PopulateContext();
+
+            ObjectResult result = (ObjectResult)_controller.Statistics(new DateTime(2020, 03, 04), new DateTime(2020, 07, 06));
+
+            Assert.AreEqual(400, result.StatusCode);
+
+            Assert.IsNotNull(result.Value);
+
+            ErrorObject error = (ErrorObject)result.Value;
+
+            Assert.AreEqual(5, error.Code);
+            Assert.AreEqual("La query non ha prodotto risultati",error.Message);
+
+        }
+
+        /**
+         * TIS-29
+         * Verificare che venga ritornato un messaggio d'errore nel caso in cui si cerchi di ottenere le statistiche con un database privo di dati
+         */
+        [TestMethod()]
+        public void EmptyDatabaseTest() {
+
+            ObjectResult result = (ObjectResult)_controller.Statistics(new DateTime(2020, 03, 04), new DateTime(2020, 07, 06));
+
+            Assert.AreEqual(400, result.StatusCode);
+
+            Assert.IsNotNull(result.Value);
+
+            ErrorObject error = (ErrorObject)result.Value;
+
+            Assert.AreEqual(5, error.Code);
+            Assert.AreEqual("La query non ha prodotto risultati", error.Message);
+
```

Comment:
```suggestion
            ErrorObject error = (ErrorObject)result.Value;
            
            Assert.AreEqual(400, result.StatusCode);
            Assert.IsNotNull(result.Value);
            Assert.AreEqual(5, error.Code);
            Assert.AreEqual("La query non ha prodotto risultati", error.Message);
```

Reply:
Dato che le suggestions non possono essere fatte direttamente da qui (indicate come "outdated") fai un commit in cui le applichi